dataprofiler/labelers/data_processing.py

```diff
@@ -598,23 +600,38 @@ def process(self, data, labels=None, label_mapping=None, batch_size=32):
         Flatten batches of data
 
         :param data: List of strings to create embeddings for
-        :type data: list() of strings
+        :type data: numpy.ndarray
         :param labels: labels for each input character
-        :type labels: list of labels for each character in data
+        :type labels: numpy.ndarray
         :param label_mapping: maps labels to their encoded integers
         :type label_mapping: Union[None, dict]
         :param batch_size: Number of samples in the batch of data
         :type batch_size: int
         :return batch_data: A dict containing  samples of size batch_size
         :rtype batch_data: dicts
         """
-        if labels is not None and not label_mapping:
-            raise ValueError('If `labels` are specified, `label_mapping` must '
-                             'also be specified.')
+        if len(data.shape) > 1:
+            # Ignore case of 1 col DataFrame
+            if not (len(data.shape) == 2 and data.shape[1] == 1):
+                raise ValueError("Multidimensional data given to "
+                                 "CharPreprocessor")
+            # Flatten 1 col DataFrame
+            data = data.reshape(-1)
+
+        if labels is not None:
+            if not label_mapping:
+                raise ValueError('If `labels` are specified, `label_mapping` '
+                                 'must also be specified.')
+            # Need to cast if input is coming from structured process
+            if isinstance(labels, list):
+                labels = np.array(labels, dtype="object")
```

Comment:
we should assume these labels are coming in as expected. the other processor should input the data into this processor as the correct type so that we don't have to make modifications for every processor that might input data into this one.